deploy_docs.sh

```diff
@@ -0,0 +1,59 @@
+# Modified from https://github.com/AMReX-Codes/amrex/blob/master/build_and_deploy.sh
+# and licensed according to BSD-3-Clause-LBNL (https://github.com/AMReX-Codes/amrex/blob/master/LICENSE).
+
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# Build the documentation from the SOURCE_BRANCH
+# and push it to TARGET_BRANCH.
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deploy)
+git clone $REPO out
+cd out
+
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+
+# Clean out existing contents
+git rm -rf . || exit 0
+cd ..
+
+# Pull from SOURCE_BRANCH again
+git pull $SSH_REPO $SOURCE_BRANCH
```

Comment:
Is that pulling again really necessary?

You come back to source branch itself by doing `cd ..` (in line 26). 
It even proves to work in my shared repos (wsynphot & starkit) - I think you should once try running pipeline without it.

Reply:
good point

Reply:
It is not necessary in this case. I did actually bring this point up before. I only left incase the source branch we are building the documentation from is not the master branch. So it is really up to the main tardis users whether they want it or not. I could then remove line 9 also. Source Branch =

Reply:
If there's any such use case, then its reasonable to keep it. :+1: